web/client/components/styleeditor/RulesEditor.jsx

```diff
@@ -253,7 +253,9 @@ const RulesEditor = forwardRef(({
                         hideScaleDenominator,
                         classificationType
                     } = ruleBlock[ruleKind] || {};
-                    const isCustomNumber =  attributes?.filter(({label}) => label === rule?.attribute)?.[0]?.type === 'number';
+
+                    // attributes could be false so it is not possible to use ?. operator
+                    const isCustomNumber =  attributes && attributes.filter(({label}) => label === rule?.attribute)?.[0]?.type === 'number';
```

Comment:
Does it makes sense the `attributes` to become false?
if attributes is **only** false or array, that this makes sense. otherwise, if it is true, it may fail again. 
I suggest to use `isArray` instead, to ensure the method filter is present, instead.

Reply:
Anyway this is strange. it is always true, isn't it?

Reply:
The PR is updated with following changes:
- return an empty array of attributes if false
- use isArray before to look inside the attributes list

This control is needed to change the disable state of the attributes options in the classification symbolizer and this state changes based on the current selected attribute type